src/Raven.Server/Integrations/PostgreSQL/PgSession.cs

```diff
@@ -193,6 +196,8 @@ await writer.WriteAsync(messageBuilder.ErrorResponse(
                     try
                     {
                         await message.Init(transaction.MessageReader, reader, _token);
+                        if (TrafficWatchManager.HasRegisteredClients && message is Query queryMessage)
+                            DispatchPostgresQueryMessageToTrafficWatch(queryMessage, nodeTag);
```

Comment:
Typically we dispatch it _after_ handling a request. This way we're able to say that either the request was handled successfully or not.

I wonder whether we shouldn't do the same here. Then you might be able to indicate an error in the traffic watch once `PgErrorException`  is thrown here.

Reply:
done